miasm/expression/simplifications_common.py

```diff
@@ -1721,11 +1721,13 @@ def simp_add_multiple(_, expr):
 
     for arg, count in viewitems(operands):
         if count == 0:
-            continue
-        if count == 1:
+            pass
+        elif count == 1:
             out.append(arg)
-            continue
-        out.append(arg * ExprInt(count, expr.size))
+        elif count < 0:
+            out.append(arg * (-ExprInt(-count, expr.size)))
```

Comment:
This may be counter productive:
The next step in simplification will be to test if we have - ExprInt(XXX) and transform this to ExprInt(-XXX)

Reply:
Indeed